Controllers/Repository.php

```diff
@@ -0,0 +1,136 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <shopware@nosto.com>
+ * @copyright Copyright (c) 2016 Nosto Solutions Ltd (http://www.nosto.com)
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ */
+
+use Shopware\Models\Category\Category;
+
+class Shopware_Plugins_Frontend_NostoTagging_Controllers_Repository
+{
+    /**
+     * Returns an array of articles id's that are active
+     * and has the same category id of the given shopware category
+     *
+     * @param \Shopware\Models\Category\Category $category
+     * @param $pageSize
+     * @param $currentOffset
+     * @param $id
+     * @return array
+     */
+    public function getActiveArticlesIdsByCategory(
+        Category $category,
+        $pageSize,
+        $currentOffset,
+        $id
+    ) {
+        $builder = Shopware()->Models()->createQueryBuilder();
+        $result = $builder->select('articles.id')
+            ->from('\Shopware\Models\Article\Article', 'articles')
+            ->innerJoin(
+                '\Shopware\Models\Article\Detail',
+                'details',
+                \Doctrine\ORM\Query\Expr\Join::WITH,
+                'articles.mainDetailId = details.id'
+            )
+            ->innerJoin(
+                'articles.categories',
+                'c'
+            )
+            ->where('articles.active = 1')
+            ->andWhere('c.path LIKE :path')
+            // Since the path in the database is saved with || between
+            // the parents ids, we concatenate those and get all child
+            // categories from the given language.
+            ->setParameter('path', '%|'.(int)$category->getId().'|%');
+        if (!empty($id)) {
+            $result = $result->andWhere('details.number = :id')
+                ->setParameter('id', $id)
+                ->getQuery();
+        } else {
+            $result = $result->orderBy('articles.added', 'DESC')
+                ->setFirstResult($currentOffset)
+                ->setMaxResults($pageSize)
+                ->getQuery();
+        }
+        return $result->getResult(\Doctrine\ORM\AbstractQuery::HYDRATE_ARRAY);
+    }
+
+    /**
+     * @param \Shopware\CustomModels\Nosto\Account\Account $account
+     * @return bool
+     */
+    public function isAccountAlreadyRegistered(
```

Comment:
This should not be in this repo. Create own helper class or repo

Reply:
Fixed